src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/MigrateVisitService.kt

```diff
@@ -102,6 +104,26 @@ class MigrateVisitService(
     return visitEntity.reference
   }
 
+  private fun getSessionCategory(migrateVisitRequest: MigrateVisitRequestDto): String? {
+    val isInTheFuture = !migrateVisitRequest.startTimestamp.isBefore(LocalDateTime.now())
+    if (isInTheFuture) {
+      val groups = sessionTemplateService.getSessionLocationGroup(migrateVisitRequest.prisonCode)
+      val sessionDate = migrateVisitRequest.startTimestamp.toLocalDate()
+      val startTime = migrateVisitRequest.startTimestamp.toLocalTime()
+      val endTime = migrateVisitRequest.endTimestamp.toLocalTime()
+
+      if (groups.isNotEmpty()) {
+        val location = prisonerService.getPrisonerHousingLocation(migrateVisitRequest.prisonerId, migrateVisitRequest.prisonCode)
+        val sessionCapacity = sessionService.getSessionCapacity(location, migrateVisitRequest.prisonCode, sessionDate, startTime, endTime)
+        return sessionCapacity?.capacityGroup
+      } else {
+        val sessionCapacities = sessionService.getSessionCapacity(migrateVisitRequest.prisonCode, sessionDate, startTime, endTime)
+        // TODO this is a list > how do we do this?
+      }
+    }
+    return null
+  }
+
```

Comment:
The above code is a tricky one, I think we should have a manual solution to this some how because migration is a pain